firmware/lib/Gantry/Axis.cpp

```diff
@@ -1,311 +1,569 @@
+/* **************************** Local Includes ***************************** */
 #include "Axis.h"
-#include "Debug.h"
-#include "Timer.h"
 #include "Movement.h"
+#include "Timer.h"
+
+/*****************************************************************************/
+/*                                  DEFINES                                  */
+/*****************************************************************************/
+
+/** TC IRQ number for x-axis velocity timer */
+#define IRQ_X_AXIS_VELOCITY    3
+/** TC IRQ number for x-axis acceleration timer */
+#define IRQ_X_AXIS_ACCEL       4
+
+/** TC IRQ number for y-axis velocity timer */
+#define IRQ_Y_AXIS_VELOCITY    0
+/** TC IRQ number for y-axis acceleration timer */
+#define IRQ_Y_AXIS_ACCEL       1
+
+/** Frequency of the slow clock */
+#define SCLK_FREQ              32768
+
+/** Minimum pulse duration to pass debouncing */
+#define DEBOUNCE_FILTER_MS     50
```

Comment:
Is this with regards to the encoder? We are waiting 50 milliseconds for what action?

Reply:
This is for debouncing the limit switches. The limit switch value has to stay steady for 50 ms to be accepted